airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -400,7 +400,7 @@ def attempt_task_runs(self):
             else:
                 task = run_task_response["tasks"][0]
                 self.active_workers.add_task(task, task_key, queue, cmd, exec_config, attempt_number)
-                self.queued(task_key, task.task_arn)
+                self.running_state(task_key, task.task_arn)
```

Comment:
Does this means that the executor wont work for airflow<2.10 ?

Reply:
It still functioned _mostly_ correctly. The running and queued task sets are maintained mostly by the base executor and used for emitting metrics and to calculate open slots. I think the regression is only present in airflow 2.9

Reply:
but what will happen if someone install newer version of the provider with Airflow 2.8.x / 2.9.0?
the call for `self.running_state` would not work doesn't it?

Reply:
Ooo, I see what you mean now, good catch! I suppose we could set a min version, but instead I'll try call that new method in a try/catch and swallow the exception. The only real consequence of that is that task adoption will not work (but this is a nice to have and many executors don't support it), the code to support that is what introduced this regression in the first place anyway. So task adoption will only work in 2.10 forward, but the rest will be backwards compatible.

Reply:
normally we backport by copying the needed function into the provider as private function with note to remove the workaround when min airflow version is greater than what is needed.

Reply:
ah maybe it's because the code doesn't import `running_state`

Reply:
Resolving this thread, put up some changes to cover the backcompat.